src/hooks/useGenres.ts

```diff
@@ -0,0 +1,48 @@
+import { IGenre } from "@globalTypes";
+import {
+  LSAPIGetFavouriteGenres,
+  LSAPIToggleFavouriteGenre,
+} from "@utils/localStorageAPI";
+import { tmdbGetGenres } from "@utils/tmdbAPI";
+import { useEffect, useState } from "react";
+import { IUseGenres } from "./types";
+
+const useGenres = (isSaveMode: boolean): IUseGenres => {
+  const [genres, setGenres] = useState<IGenre[]>([]);
+
+  useEffect(() => {
+    tmdbGetGenres()
+      .then((data) => {
+        const getedGenres = data.genres;
+        const favoriteIdx = LSAPIGetFavouriteGenres();
+        getedGenres.map(
+          (genre) => (genre.isFavourite = favoriteIdx.includes(genre.id))
+        );
+        setGenres(getedGenres);
+      })
+      .catch((error: Error) =>
+        console.log("Genre list loading error: ", error)
+      );
+  }, []);
+
+  const toggleFavouriteGenre = (genreId: number): void => {
+    const newGenres = [...genres];
+    const foundGenre = genres.find((genre) => genre.id === genreId);
+    const { isFavourite } = foundGenre!;
```

Comment:
Same here, please avoid of non-null assertion operator. 

1. If you're 100% sure that genre always should be, then use `invariant` (or better `tiny-invariant`).
2. If you allow the possibility, then `if (foundGenre == null) return`